application/src/main/java/org/togetherjava/tjbot/features/jshell/JShellCommand.java

```diff
@@ -0,0 +1,299 @@
+        }
+    }
+
+    private void sendEvalModal(SlashCommandInteractionEvent event, boolean startupScript) {
+        TextInput body = TextInput
+            .create(TEXT_INPUT_PART_ID + (startupScript ? "|" + STARTUP_SCRIPT_PARAMETER : ""),
+                    "Enter code to evaluate.", TextInputStyle.PARAGRAPH)
+            .setPlaceholder("Put your code here.")
+            .setRequiredRange(MIN_MESSAGE_INPUT_LENGTH, MAX_MESSAGE_INPUT_LENGTH)
+            .build();
+
+        Modal modal = Modal.create(generateComponentId(), "JShell").addActionRow(body).build();
+        event.replyModal(modal).queue();
+    }
+
+    /**
+     * Handle evaluation of code.
+     *
+     * @param replyCallback the callback to reply to
+     * @param user the user, if null, will create a single use session
+     * @param showCode if the embed should contain the original code
+     * @param startupScript if the startup script should be used or not
+     * @param code the code
+     */
+    private void handleEval(IReplyCallback replyCallback, @Nullable User user, boolean showCode,
+            String code, boolean startupScript) {
+        replyCallback.deferReply().queue(interactionHook -> {
+            try {
+                interactionHook
+                    .editOriginalEmbeds(
+                            jshellEval.evaluateAndRespond(user, code, showCode, startupScript))
+                    .queue();
+            } catch (RequestFailedException e) {
+                interactionHook.editOriginalEmbeds(createUnexpectedErrorEmbed(user, e)).queue();
+            }
+        });
+    }
+
+    private void handleSnippetsCommand(SlashCommandInteractionEvent event) {
+        event.deferReply().queue(interactionHook -> {
+            OptionMapping userOption = event.getOption(USER_PARAMETER);
+            User user = userOption == null ? event.getUser() : userOption.getAsUser();
+            OptionMapping includeStartupScriptOption =
+                    event.getOption(INCLUDE_STARTUP_SCRIPT_PARAMETER);
+            boolean includeStartupScript =
+                    includeStartupScriptOption != null && includeStartupScriptOption.getAsBoolean();
+            List<String> snippets;
+            try {
+                snippets = jshellEval.getApi()
+                    .snippetsSession(user.getId(), includeStartupScript)
+                    .snippets();
+            } catch (RequestFailedException e) {
+                if (e.getStatus() == JShellApi.SESSION_NOT_FOUND) {
+                    interactionHook.editOriginalEmbeds(createSessionNotFoundErrorEmbed(user))
+                        .queue();
+                } else {
+                    interactionHook.editOriginalEmbeds(createUnexpectedErrorEmbed(user, e)).queue();
+                }
+                return;
+            }
+
+            if (snippets.stream().noneMatch(s -> s.length() >= MessageEmbed.VALUE_MAX_LENGTH)
```

Comment:
why not splitting the logic into private methods?
so the `if` doesn't  will not be this clumped up

Reply:
Changed, tell me what you think.

Reply:
I didnt really mean the if itself, I meant to have a `private boolean ...` method for `snippets.stream().noneMatch(s -> s.length() >= MessageEmbed.VALUE_MAX_LENGTH)`, `snippets.stream().mapToInt(s -> (s + "Snippet 10```java\n```").length()).sum() < MessageEmbed.EMBED_MAX_LENGTH_BOT - 100`, ...

Reply:
How is it now ?

Reply:
yeah I am fine with that